vzstorage-pd/vendor/github.com/virtuozzo/goploop-cli/ploop-volume.go

```diff
@@ -0,0 +1,71 @@
+package ploop
+
+import (
+	"fmt"
+	"os"
+	"path"
+)
+
+var (
+	ploopVolumeBinary = "/usr/sbin/ploop-volume"
+)
+
+type PloopVolume struct {
+	Path string
+}
+
+func PloopVolumeOpen(src string) (*PloopVolume, error) {
+	if _, err := os.Stat(path.Join(src, "DiskDescriptor.xml")); os.IsNotExist(err) {
+		return nil, &Err{c:-1, s: fmt.Sprintf("Bad ploop-volume path %s!", src)}
+	}
+	return &PloopVolume{src}, nil
+}
+
+func PloopVolumeCreate(src string, size uint64, image string) (*PloopVolume, error) {
+	if _, err := os.Stat(src); !os.IsNotExist(err) {
+		return nil, &Err{c:-1, s: fmt.Sprintf("Destination ploop directory already exists %s!", src)}
+	}
+
+	args := []string{"create", "-s", fmt.Sprintf("%v", size)}
+	if image != "" {
+		args = append(args, "--image", image)
+	}
+	args = append(args, src)
+	if err := ploopVolume(args...); err != nil {
+		return nil, err
+	}
+	return &PloopVolume{src}, nil
+}
+
+func (pv *PloopVolume) Snapshot(dst string) error {
+	if dst == "" {
+		return &Err{c:-1, s: fmt.Sprintf("Bad destination path!")}
+	}
+	return ploopVolume("snapshot", pv.Path, dst)
+}
+
+func (pv *PloopVolume) Switch(dst string) error {
+	if dst == "" {
+		return &Err{c:-1, s: fmt.Sprintf("Bad destination path!")}
+	}
+	if _, err := os.Stat(pv.Path); os.IsNotExist(err) {
```

Comment:
Actually we can deal without it, but it makes this function a little bit more robust. Depends on you, I can easily remove it